repos/system_upgrade/el9toel10/actors/mysqlcheck/actor.py

```diff
@@ -0,0 +1,21 @@
+from leapp.actors import Actor
+from leapp.libraries.actor.mysqlcheck import MySQLCheckLib
+from leapp.models import DistributionSignedRPM, Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+
+class MySQLCheck(Actor):
+    """
+    Actor checking for presence of MySQL installation.
+
+    Provides user with information related to upgrading systems
+    with MySQL installed.
+    """
+    name = 'mysql_check'
+    consumes = (DistributionSignedRPM,)
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
```

Comment:
The actor is interacting with the system. It needs to be split in two actors:
* scanner scanner - that will collect all important data (facts). Based on the actor code, it seems you want a Model like this (I expect that most likely you will want to change a little bit):
```
class MySQLConfiguration(Model):
    """
    describe the content of the model
    """
    
    removed_options = fields.List(fields.String, default=[])
    """
    Configured options which are removed in RHEL 10 MySQL
    """
    
    systemd_services = fields.List(fields.String, default=[])
    """
    ...
    """
```
* the checker - that consumes particular leapp message(s) and evaluate the situation.